src/FromFile.jl

```diff
@@ -12,23 +13,62 @@ function __init__()
     @require Revise="295af30f-e4ad-537b-8983-00126c2a3abe" track(mod, path) = Revise.track(mod, path)
 end
 
-macro from(path::String, ex::Expr)
+macro from(path, ex::Expr)
     esc(from_m(__module__, __source__, path, ex))
 end
 
-macro from(path::String)
-    esc(from_m(__module__, __source__, path, Expr(:block))) 
+macro from(path)
+    esc(from_m(__module__, __source__, path))
 end
 
-function from_m(m::Module, s::LineNumberNode, path::String, root_ex::Expr)
-    import_exs = if root_ex.head === :block
-        filter(ex -> !(ex isa LineNumberNode), root_ex.args)
-    else
-        [root_ex]
+macro from_url(path, ex::Expr)
+    esc(from_url_m(__module__, __source__, path, ex))
+end
+
+macro from_url(path)
+    esc(from_url_m(__module__, __source__, path))
+end
+
+# NOTE: we'd like to keep this pacakge minimal deps
+# so just copy paste this one-liner
+# Source: IsUrl.jl (MIT license)
+function isurl(str::AbstractString)
+    windowsregex = r"^[a-zA-Z]:[\\]"
+    urlregex = r"^[a-zA-Z][a-zA-Z\d+\-.]*:"
+    return !occursin(windowsregex, str) && occursin(urlregex, str)
+end
+
+function from_m(m::Module, s::LineNumberNode, path, root_ex::Expr = Expr(:block))
+    path = eval_quoted_string_expr(m, path)
+    return from_local_file(m, s, path, root_ex)
+end
+
+function from_url_m(m::Module, s::LineNumberNode, url, root_ex::Expr = Expr(:block))
+    url = eval_quoted_string_expr(m, url)
+    isurl(url) || throw(ArgumentError("expects an URL"))
```

Comment:
We probably shouldn't check this ourselves. Better to just let `Downloads.download` handle it, and raise an error if a malformed URL is passed.

Reply:
I think `download`'s error msg is less intuitive in this sense, it's better to throw the error earlier with a clear sentence about what we are expecting.